Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,480 @@
+        {
+            try
+            {
+                ObservableList<AppModelParameter> lstParameters = null;
+                ObservableList<ActInputValue> actInputs = ((ActWebAPIBase)act).DynamicElements;
+                if (actInputs == null || actInputs.Count == 0)
+                {
+                    string requestBody = string.Empty;
+                    if (aPIModel.RequestBodyType == ApplicationAPIUtils.eRequestBodyType.TemplateFile)
+                    {
+                        string fileUri = WorkSpace.Instance.SolutionRepository.ConvertSolutionRelativePath(aPIModel.TemplateFileNameFileBrowser);
+                        if (File.Exists(fileUri))
+                        {
+                            requestBody = System.IO.File.ReadAllText(fileUri);
+                        }
+                    }
+                    else
+                    {
+                        if (string.IsNullOrEmpty(aPIModel.RequestBody))
+                        {
+                            SetPropertyValue(aPIModel, nameof(ActWebAPIBase.Fields.RequestBody), nameof(ApplicationAPIModel.RequestBody), act);
+                            requestBody = Convert.ToString(aPIModel.RequestBody);
+                        }
+                        else
+                        {
+                            requestBody = Convert.ToString(aPIModel.RequestBody);
+                        }
+                    }
+
+                    ObservableList<ApplicationAPIModel> applicationAPIModels = new ObservableList<ApplicationAPIModel>();
+                    if (!string.IsNullOrEmpty(requestBody))
+                    {
+                        if (aPIModel.RequestBody.StartsWith("{"))
+                        {
+                            JSONTemplateParser jsonTemplate = new JSONTemplateParser();
+                            jsonTemplate.ParseDocumentWithJsonContent(requestBody, applicationAPIModels);
+                        }
+                        else if (aPIModel.RequestBody.StartsWith("<"))
+                        {
+                            XMLTemplateParser parser = new XMLTemplateParser();
+                            parser.ParseDocumentWithXMLContent(requestBody, applicationAPIModels);
+                        }
+
+                        if (!isModelExists && applicationAPIModels[0].AppModelParameters != null && applicationAPIModels[0].AppModelParameters.Count > 0)
+                        {
+                            aPIModel.AppModelParameters = applicationAPIModels[0].AppModelParameters;
+                        }
+                    }
+
+                    if (!isModelExists && parameterizeRequestBody && applicationAPIModels.Count > 0 && !string.IsNullOrEmpty(applicationAPIModels[0].RequestBody))
+                    {
+                        aPIModel.RequestBody = applicationAPIModels[0].RequestBody;
+                    }
+
+                    if (!isModelExists)
```

Comment:
i'm confuzed, that mean that you don't learn optional value in case API model exist? but that we should do also for existing API model

Reply:
ok changed the condition